src/client/resources/web/js/message_parsing.mjs

```diff
@@ -31,37 +31,233 @@ const VALID_COLORS = [
     'reset'
 ];
 
+const VALID_HOVER_EVENTS = [
+    'show_text',
+    'show_item',
+    'show_entity'
+];
+
 /**
- * @typedef {Object} BaseComponent
+ * @typedef {Object} Component
  * @property {string} [text] - Text content
  * @property {string} [translate] - Translation key
- * @property {Array<string|Component>} [with] - Translation parameters
- * @property {Array<Component>} [extra] - Additional components to append
+ * @property {Component[]} [with] - Translation parameters
+ * @property {(string | Component)[]} [extra] - Additional components to append
  * @property {string} [color] - Text color - can be a named color or hex value
  * @property {boolean} [bold] - Whether text should be bold
  * @property {boolean} [italic] - Whether text should be italic
  * @property {boolean} [underlined] - Whether text should be underlined
  * @property {boolean} [strikethrough] - Whether text should be struck through
  * @property {boolean} [obfuscated] - Whether text should be obfuscated (randomly changing characters)
+ * @property {HoverEvent} [hoverEvent] - Hover event
+ */
+
+/**
+ * @typedef {ShowTextHoverEvent | ShowItemHoverEvent | ShowEntityHoverEvent} HoverEvent
+ */
+
+/**
+ * @typedef {Object} ShowTextHoverEvent
+ * @property {'show_text'} action - Displays a text tooltip
+ * @property {string | Component | (string | Component)[]} contents - The text content to show in the tooltip
+ */
+
+/**
+ * @typedef {Object} ShowItemHoverEvent
+ * @property {'show_item'} action - Displays an item's tooltip
+ * @property {{ id: string, count?: number, tag?: string }} contents - The item data to show
  */
 
 /**
- * @typedef {BaseComponent & {
- *   name: string
- * }} HoverEventContents
+ * @typedef {Object} ShowEntityHoverEvent
+ * @property {'show_entity'} action - Displays entity information
+ * @property {{ type: string, id: unknown, name?: string }} contents - The entity data to show
```

Comment:
I left `id` as `unknown`. It seems it should be a UUID string, but it's coming through as an array of 4 numbers (which could make sense if the 4 numbers are 1:1 with a UUID at 32 bits each). So perhaps we're doing a different serialization than the Wiki expects.

Reply:
Oh btw 

> So perhaps we're doing a different serialization than the Wiki expects.

We are using native serialization, its also possible the wiki is incorrect. Which wiki have you been looking at?